types/v56/types.go

```diff
@@ -1818,6 +1818,27 @@ type LbVirtualServer struct {
 	ApplicationRuleIds   []string `xml:"applicationRuleId,omitempty"`
 }
 
+// EdgeNatRule contains shared structure for SNAT and DNAT rule configuration using
+// NSX-V proxied edge gateway endpoint
+// // https://code.vmware.com/docs/6900/vcloud-director-api-for-nsx-programming-guide
+type EdgeNatRule struct {
+	XMLName           xml.Name `xml:"natRule"`
+	ID                string   `xml:"ruleId,omitempty"`
+	RuleType          string   `xml:"ruleType,omitempty"`
+	RuleTag           string   `xml:"ruleTag,omitempty"`
+	Action            string   `xml:"action,omitempty"`
+	Vnic              *int     `xml:"vnic,omitempty"`
+	OriginalAddress   string   `xml:"originalAddress,omitempty"`
+	TranslatedAddress string   `xml:"translatedAddress,omitempty"`
```

Comment:
Is it really all fields can be empty?

Reply:
Added just a few.

Reply:
In general we match API description, which are mandatory, which aren't. It isn't decision from our point of view.

Reply:
Yes, but only these are mandatory. NSX can default everything else.